Automattic/delphin
# Upgrade to react to 15.4.0+

React changed their internal API in 15.4.0, and `react-tap-event-plugin` depended on it. This patch updates `react-tap-event-plugin` to `2.0.0` to fix that. See https://github.com/zilverline/react-tap-event-plugin/issues/85 for more info

```
ERROR in ./~/react-tap-event-plugin/src/injectTapEventPlugin.js
Module not found: Error: Can't resolve 'react/lib/EventPluginHub' in '/home/ubuntu/data/gh-delphin/branches/fix/ie-bugs/node_modules/react-tap-event-plugin/src'
 @ ./~/react-tap-event-plugin/src/injectTapEventPlugin.js 23:2-37
 @ ./client/index.js
 @ multi app

ERROR in ./~/react-tap-event-plugin/src/TapEventPlugin.js
Module not found: Error: Can't resolve 'react/lib/EventConstants' in '/home/ubuntu/data/gh-delphin/branches/fix/ie-bugs/node_modules/react-tap-event-plugin/src'
 @ ./~/react-tap-event-plugin/src/TapEventPlugin.js 22:21-56
 @ ./~/react-tap-event-plugin/src/injectTapEventPlugin.js
 @ ./client/index.js
 @ multi app

ERROR in ./~/react-tap-event-plugin/src/TapEventPlugin.js
Module not found: Error: Can't resolve 'react/lib/EventPluginUtils' in '/home/ubuntu/data/gh-delphin/branches/fix/ie-bugs/node_modules/react-tap-event-plugin/src'
 @ ./~/react-tap-event-plugin/src/TapEventPlugin.js 23:23-60
 @ ./~/react-tap-event-plugin/src/injectTapEventPlugin.js
 @ ./client/index.js
 @ multi app

ERROR in ./~/react-tap-event-plugin/src/TapEventPlugin.js
Module not found: Error: Can't resolve 'react/lib/EventPropagators' in '/home/ubuntu/data/gh-delphin/branches/fix/ie-bugs/node_modules/react-tap-event-plugin/src'
 @ ./~/react-tap-event-plugin/src/TapEventPlugin.js 24:23-60
 @ ./~/react-tap-event-plugin/src/injectTapEventPlugin.js
 @ ./client/index.js
 @ multi app

ERROR in ./~/react-tap-event-plugin/src/TapEventPlugin.js
Module not found: Error: Can't resolve 'react/lib/SyntheticUIEvent' in '/home/ubuntu/data/gh-delphin/branches/fix/ie-bugs/node_modules/react-tap-event-plugin/src'
 @ ./~/react-tap-event-plugin/src/TapEventPlugin.js 25:23-60
 @ ./~/react-tap-event-plugin/src/injectTapEventPlugin.js
 @ ./client/index.js
 @ multi app

ERROR in ./~/react-tap-event-plugin/src/TapEventPlugin.js
Module not found: Error: Can't resolve 'react/lib/ViewportMetrics' in '/home/ubuntu/data/gh-delphin/branches/fix/ie-bugs/node_modules/react-tap-event-plugin/src'
 @ ./~/react-tap-event-plugin/src/TapEventPlugin.js 27:22-58
 @ ./~/react-tap-event-plugin/src/injectTapEventPlugin.js
 @ ./client/index.js
 @ multi app
```

A better fix would be to remove this dependency from our project since it doesn't seem to fix anything on new browsers anyway https://developers.google.com/web/updates/2013/12/300ms-tap-delay-gone-away.

### Testing Instructions
- `git checkout fix/react-tap-event-plugin`
- `npm run clean:all`
- `npm install`
- `npm run build`
Check that it works

### Reviews
